site/docs/extensions/index.md

```diff
@@ -0,0 +1,38 @@
+
+A Substrait plan can reference one or more YAML files via URI for extension. In the places where these entities are referenced, they will be referenced using a URI + name reference. The name scheme per type works as follows:
+
+| Category            | Naming scheme                                                |
+| ------------------- | ------------------------------------------------------------ |
+| Types               | The name as defined on the type object.                      |
+| Type Variations     | The name as defined on the type object.                      |
+| Function Signatures | If there is only one impl, the name may be defined at the top-level definition of the object. If there are multiple impls, the name is a composite string that includes the top-level name (e.g. add) followed by a compound string that describes the types of the arguments (e.g. add:i8_i8). In the case there is one impl, the multiple impl naming scheme is also supported. How arguments are mapped to a compound name are described below. |
+
+## Advanced Extensions
+
+Less common extensions can be extended using customization support at the serialization level. This includes the following kinds of extensions:
+
+| Extension Type                           | Description                                                  |
+| ---------------------------------------- | ------------------------------------------------------------ |
+| Semantic Changing Relation Extensions    | Extensions to an existing relation that will alter the semantics of that relation. These kinds of extensions require that any plan consumer understand the extension to be able to manipulate or execute that operator. Ignoring these extensions will result in an incorrect interpretation of the plan. An example extension might be creating a customized version of Aggregate that can optionally apply a filter before aggregating the data. <br /><br />Note: Semantic-changing extensions shouldn't change the core characteristics of the underlying relation. For example, they should *not* change the default direct output field ordering, change the number of fields output or change the behavior of physical property characteristics. If one needs to change one of these behaviors, one should define a new relation as described below. |
+| Optimization-Related Relation Extensions | Extensions to an existing relation that can improve the efficiency of a plan consumer but don't fundamentally change the behavior of the operation. An example might be an estimated amount of memory the relation is expected to use or a particular algorithmic pattern that is perceived to be optimal. |
```

Comment:
The terms "Semantic Changing Relation Extensions" and "Optimization-Related Relation Extensions" are a lot to swallow.  Also, the other items in this table don't end with "Extensions".  Maybe "Relation Modification (semantic)" and "Relation Modification (optimization)"?

Reply:
I'll do another pass on this.

Reply:
Updated